### PR TITLE
LibRegex: Fail early if searching anchored patterns

### LINKED ISSUE #700: LibRegex: Searching for anchored patterns like /^b/ should fail early if possible
If we're looking for `/^b/` in the string `aaaaaa` (and multi-line mode is not enabled), we should be smarter than to scan through every single `a` before bailing.

Reduction from https://fragrantica.com/

```js
const mb = 100;
console.log("Create " + mb + "MB string");
let s = "a".repeat(mb * 1048576);

let re = /^b/
console.log("Search for " + re + " in string");
for (let i = 0; i < 10_000; ++i) {
    console.log(re.test(s));
}
```

## Funding

- We're using [Polar.sh](https://polar.sh/LadybirdBrowser) to allow folks to offer a reward for completing this issue.
- The contributor(s) receive the funding once the issue is completed & confirmed.

<a href="https://polar.sh/LadybirdBrowser/ladybird/issues/700">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/LadybirdBrowser/ladybird/issues/700/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by ChaseKnowlden**
This fixes #700 

Check if searching for anchored patterns fails early.

### CODE REVIEW

File: Userland/Libraries/LibRegex/RegexParser.cpp [hunk 1]
```diff
@@ -372,6 +373,8 @@ bool PosixBasicParser::parse_root(ByteCode& bytecode, size_t& match_length_minim
     if (match(TokenType::Circumflex)) {
         consume();
         bytecode.empend((ByteCodeValueType)OpCodeId::CheckBegin);
+    } else {
+        return false;
```
Comment (alimpfard) [inline Userland/Libraries/LibRegex/RegexParser.cpp:377]: This is most likely not what you intended to do (unless you intended to make every posix basic regexp invalid unless it starts with `^`?)

**Comment by alimpfard** [comment]
The issue you've referenced is about the ECMA262 flavour, but generally that has nothing to do with the parsing stage; what you want to do is to find a fast path for `CheckBegin` to fail the entire match (or as much as it can, ideally) instead of the match starting from "the current position".

**Comment by ChaseKnowlden** [comment]
CI is passing!

**Comment by Gingeh** [comment]
This doesn't seem to have sped up the example in #700

**Comment by ChaseKnowlden** [comment]
Closing as wontfix.

**Pull request closed without merging**